lib/utilities/monitoringHandler.js

```diff
@@ -80,10 +92,11 @@ function promMetrics(requestType, bucketName, code, typeOfRequest,
     && code === '200') {
         dataDiskAvailable.inc(newByteLength);
         dataDiskFree.inc(newByteLength);
-        if (numOfObjectsRemoved) {
-            numberOfObjects.dec(numOfObjectsRemoved);
-        } else {
-            numberOfObjects.dec();
+        const objs = numOfObjectsRemoved || 1;
```

Comment:
is it still correct if the Put Object is not a replacement but a creation?

Reply:
The behavior is the same as before. `dec()` without a value defaults to 1. In this case it only does this for `deleteObject`, `abortMultipartUpload`, `multiObjectDelete` operations so the counter should remain correct.